src/Stratis.Bitcoin.Features.BlockStore/BlockRepository.cs

```diff
@@ -58,55 +68,81 @@ public interface IBlockRepository : IBlockStore
         /// <param name="txIndex">Whether to index transactions.</param>
         Task SetTxIndexAsync(bool txIndex);
 
+        /// <summary>Hash and height indicating where the node's block store has been pruned up to.</summary>
+        HashHeightPair PrunedTip { get; }
```

Comment:
@noescape00 are you suggesting that I move ALL of the  prune related properties and methods to `PrunedBlockRepository`? In other words create a new class that inherits from `BlockRepository`?

Reply:
>I think it will be better to inherit from BlockRepository and do IPrunedBlockRepository

I am not sure I agree, because pruning is a property also of the node itself (not just dbreeze)
If we use inheritance it means in addition to stratisd will have prunedStratisD (or we will have to check the pruned flag and replace the implementation on startup

Reply:
Yeah IMO its better to have this logic separate from the BlockRepository so that we can test and configure the node easier.